overlord/snapstate/snapmgr.go

```diff
@@ -121,8 +121,18 @@ type SnapSetup struct {
 	// that it wasn't create in the current change.
 	MigratedToExposedHome bool `json:"migrated-exposed-home,omitempty"`
 
+	// RemovedExposedHome is set if the ~/Snap sub directory was removed. This
+	// should only happen when undoing the creation of that directory in the same
+	// (failed) change. To disable usage of the exposed home in a change after it
+	// was created, SnapSetup.DisableExposedHome should be used.
+	RemovedExposedHome bool `json:"removed-exposed-home,omitempty"`
+
+	// EnableExposedHome is set if the ~/Snap sub directory already exists and
+	// should be used.
+	EnableExposedHome bool `json:"enable-exposed-home,omitempty"`
```

Comment:
This flag isn't set yet but I think adding it now makes the undo helpers more symmetric and logical. This will be set once we add the forward path to the snap revert (which is one of the next steps).